src/controller/proxy/controller.go

```diff
@@ -53,13 +54,13 @@ type Controller interface {
 	// UseLocalBlob check if the blob should use local copy
 	UseLocalBlob(ctx context.Context, art lib.ArtifactInfo) bool
 	// UseLocalManifest check manifest should use local copy
-	UseLocalManifest(ctx context.Context, art lib.ArtifactInfo) (bool, error)
+	UseLocalManifest(ctx context.Context, art lib.ArtifactInfo, remote RemoteInterface) (bool, error)
```

Comment:
I think the `RemoteInterface` can be a structure property of the `Controller` implementation rather than passing as a parameter of the function?

Reply:
Because the controller is a singleton and initialize only once, it cannot handle the case when there more than 2 proxy servers.

Reply:
The `RemoteInterface` is created inside the function implementation for `ProxyBlob`: https://github.com/goharbor/harbor/blob/master/src/controller/proxy/controller.go#L156

Can we keep all of them in a consistent way? 
Personally, I think creating it inside the function is friendly than passing as a parameter for callers

Reply:
keep the current impl to reduce the create of remoteInterface.